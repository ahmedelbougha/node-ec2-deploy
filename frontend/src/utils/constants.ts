import placeHolderImage from '../assets/images/placeholder.png';

export const theme = {
  colors: {
    header: {
      backgroundColor: '#00264d',
      color: '#fff',
    },
    body: {
      backgroundColor: '#e3fffe',
      color: '#00264d',
    },
    footer: {
      backgroundColor: '#00264d',
      color: '#fff',
    },
    card: {
      backgroundColor: '#fbfbfb',
      color: '#000',
    },
  },
  placeHolderImage,
};

export const EventTypes: {
  [key: string]: {
    color: string;
    background: string;
    iconStyle: string;
    extraData: string[];
  };
} = {
  alert_qualified: {
    color: '#fff',
    background: '#961e1e',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  alert_raised: {
    color: '#fff',
    background: '#fa1e1e',
    iconStyle: '#1e96f3',
    extraData: ['alert_severity'],
  },
  catheter_observation: {
    color: '#fff',
    background: '#de6cec',
    iconStyle: '#1e96f3',
    extraData: ['volume_ml'],
  },
  check_in: {
    color: '#fff',
    background: '#01d5aa',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  check_out: {
    color: '#fff',
    background: '#cacf9e',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  concern_raised: {
    color: '#fff',
    background: '#8d528a',
    iconStyle: '#1e96f3',
    extraData: ['severity'],
  },
  fluid_intake_observation: {
    color: '#fff',
    background: '#87bef7',
    iconStyle: '#1e96f3',
    extraData: ['consumed_volume_ml'],
  },
  food_intake_observation: {
    color: '#fff',
    background: '#201cc4',
    iconStyle: '#1e96f3',
    extraData: ['meal', 'note'],
  },
  general_observation: {
    color: '#fff',
    background: '#0e699f',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  incontinence_pad_observation: {
    color: '#fff',
    background: '#ee1f4f',
    iconStyle: '#1e96f3',
    extraData: ['pad_condition'],
  },
  medication_schedule_created: {
    color: '#fff',
    background: '#29a88e',
    iconStyle: '#1e96f3',
    extraData: ['note', 'type', 'dose_size'],
  },
  medication_schedule_updated: {
    color: '#fff',
    background: '#dca534',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  mental_health_observation: {
    color: '#fff',
    background: '#b5df79',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  mood_observation: {
    color: '#fff',
    background: '#5d6a9f',
    iconStyle: '#1e96f3',
    extraData: ['mood'],
  },
  no_medication_observation_received: {
    color: '#fff',
    background: '#91e2e7',
    iconStyle: '#1e96f3',
    extraData: ['medication_type', 'expected_dose_timestamp'],
  },
  physical_health_observation: {
    color: '#fff',
    background: '#322c6b',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  regular_medication_maybe_taken: {
    color: '#fff',
    background: '#506c57',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  regular_medication_not_taken: {
    color: '#fff',
    background: '#698bd5',
    iconStyle: '#1e96f3',
    extraData: ['note', 'medication_type', 'medication_failure_reason'],
  },
  regular_medication_partially_taken: {
    color: '#fff',
    background: '#ec8dda',
    iconStyle: '#1e96f3',
    extraData: ['note', 'medication_type'],
  },
  regular_medication_taken: {
    color: '#fff',
    background: '#398c53',
    iconStyle: '#1e96f3',
    extraData: ['medication_type'],
  },
  task_completed: {
    color: '#fff',
    background: '#28b5bd',
    iconStyle: '#1e96f3',
    extraData: ['task_schedule_note', 'task_definition_description'],
  },
  task_completion_reverted: {
    color: '#fff',
    background: '#23b798',
    iconStyle: '#1e96f3',
    extraData: ['task_schedule_note', 'task_definition_description'],
  },
  task_schedule_created: {
    color: '#fff',
    background: '#ff856f',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  toilet_visit_recorded: {
    color: '#fff',
    background: '#e2a6ea',
    iconStyle: '#1e96f3',
    extraData: ['visit_type', 'visit_count', 'note', 'observed'],
  },
  visit_cancelled: {
    color: '#fff',
    background: '#a5e7fc',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  visit_completed: {
    color: '#fff',
    background: '#4bdcbc',
    iconStyle: '#1e96f3',
    extraData: [],
  },
};
