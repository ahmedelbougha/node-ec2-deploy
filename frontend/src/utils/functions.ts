import { EventTypes } from './constants';

export function buildEventTableEvents() {
  let eventTypes = [];

  // get the main keys of event types in order to use them to display events counts
  // in the EventTable
  const eventTypeKeys = Object.keys(EventTypes);

  for (let i = 0; i < eventTypeKeys.length; i += 2) {
    // build eventTypes array with 2 elements at a time
    // to build the EventTable as 2 events with their count per row
    eventTypes.push([eventTypeKeys[i], eventTypeKeys[i + 1]]);
  }

  return eventTypes;
}
