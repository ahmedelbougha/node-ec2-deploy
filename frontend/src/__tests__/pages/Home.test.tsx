import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../../utils/constants';
import HomePage from '../../pages/Home';

const router = createBrowserRouter([
  {
    // home page
    path: '/',
    element: <HomePage />,
  },
]);

const mockStore = configureStore([]);

describe('Home Page', () => {
  let testStore: any;

  beforeEach(() => {
    testStore = mockStore({
      recipients: {
        data: [
          {
            care_recipient_id: 'df50cac5-293c-490d-a06c-ee26796f850d',
          },
          {
            care_recipient_id: 'e3e2bff8-d318-4760-beea-841a75f00227',
          },
          {
            care_recipient_id: 'ad3512a6-91b1-4d7d-a005-6f8764dd0111',
          },
        ],
      },
    });
  });

  it('Home page in initial state', () => {
    testStore = mockStore({
      recipients: {
        data: [],
        summaryRecipient: {},
        eventsRecipient: [],
      },
    });
    const component = renderer.create(
      <Provider store={testStore}>
        <RouterProvider router={router} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home page after loading recipients', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Provider store={testStore}>
          <RouterProvider router={router} />
        </Provider>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
