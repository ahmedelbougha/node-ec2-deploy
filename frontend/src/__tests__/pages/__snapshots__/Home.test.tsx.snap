// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Home page after loading recipients 1`] = `
Array [
  <h2>
    Welcome to Birdie customer portal.
  </h2>,
  <div
    className="sc-bcXHqe gXVBWp"
  >
    <div
      className="sc-eDvSVe iXQdcB"
    >
      <div>
        <img
          alt="Mr./Mrs. df50c"
          src="placeholder.png"
        />
      </div>
      <div>
        <h2>
          Mr./Mrs. df50c
        </h2>
        <div
          className="sc-hLBbgP geNomd"
        />
        <p>
          Birdie Client
        </p>
        <a
          href="/dashboard/df50cac5-293c-490d-a06c-ee26796f850d"
          onClick={[Function]}
        >
          <button
            className="sc-dkrFOg gARiqw"
          >
             Details 
          </button>
        </a>
      </div>
    </div>
    <div
      className="sc-eDvSVe iXQdcB"
    >
      <div>
        <img
          alt="Mr./Mrs. e3e2b"
          src="placeholder.png"
        />
      </div>
      <div>
        <h2>
          Mr./Mrs. e3e2b
        </h2>
        <div
          className="sc-hLBbgP geNomd"
        />
        <p>
          Birdie Client
        </p>
        <a
          href="/dashboard/e3e2bff8-d318-4760-beea-841a75f00227"
          onClick={[Function]}
        >
          <button
            className="sc-dkrFOg gARiqw"
          >
             Details 
          </button>
        </a>
      </div>
    </div>
    <div
      className="sc-eDvSVe iXQdcB"
    >
      <div>
        <img
          alt="Mr./Mrs. ad351"
          src="placeholder.png"
        />
      </div>
      <div>
        <h2>
          Mr./Mrs. ad351
        </h2>
        <div
          className="sc-hLBbgP geNomd"
        />
        <p>
          Birdie Client
        </p>
        <a
          href="/dashboard/ad3512a6-91b1-4d7d-a005-6f8764dd0111"
          onClick={[Function]}
        >
          <button
            className="sc-dkrFOg gARiqw"
          >
             Details 
          </button>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Home Page Home page in initial state 1`] = `
<div
  className="sc-gswNZR kkvFUJ"
>
  <h2>
    There's no care recipients records in the time being!
  </h2>
</div>
`;
